integrations/wpseo/wpseo.php

```diff
@@ -404,25 +405,56 @@ public function wpseo_canonical( $url ) {
 	public function frontend_presentation( $presentation ) {
 		if ( is_front_page() ) {
 			$presentation->model->permalink = pll_home_url();
+			$presentation->model->title = WPSEO_Options::get( 'title-home-wpseo' );
+			$presentation->model->description = WPSEO_Options::get( 'title-home-wpseo' );
```

Comment:
As we see together during your explanations, only testing is_front_page is too restrictive because we make the replacement only for a static front page and not for the blog posts page.